Assets/MixedRealityToolkit/Inspectors/Setup/BuildDeployWindow.cs

```diff
@@ -0,0 +1,1436 @@
+
+        private readonly GUIContent ExecuteOnAllDevicesLabel = new GUIContent("Execute action on all devices", "Should the build options perform actions on all the connected devices?");
+
+        private readonly GUIContent AlwaysUninstallLabel = new GUIContent("Always uninstall before install", "Uninstall application before installing");
+
+        private readonly GUIContent ResearchModeCapabilityLabel = new GUIContent("Enable Research Mode", "Enables research mode of HoloLens. This allows access to raw sensor data.");
+
+        private readonly GUIContent AllowUnsafeCodeLabel = new GUIContent("Allow Unsafe Code", "Modify 'Assembly-CSharp.csproj' to allow use of unsafe code. Be careful using this in production.");
+
+        private readonly GUIContent RefreshBuildsLabel = new GUIContent("Refresh Builds", "Re-scan build directory for Appx Packages that can be deployed.");
+
+        private readonly GUIContent InstallAppXLabel = new GUIContent("Install AppX", "Install listed AppX item to either currently selected device or all devices.");
+
+        private readonly GUIContent UninstallAppXLabel = new GUIContent("Uninstall AppX", "Uninstall listed AppX item to either currently selected device or all devices.");
+
+        private readonly GUIContent LaunchAppLabel = new GUIContent("Launch App", "Launch listed app on either currently selected device or all devices.");
+
+        private readonly GUIContent ViewPlayerLogLabel = new GUIContent("View Player Log", "Launch notepad with more recent player log for listed AppX on either currently selected device or from all devices.");
+
+        #endregion Labels
+
+        #region Properties
+
+        private static bool IsValidSdkInstalled { get; set; } = true;
+
+        private static bool ShouldOpenSLNBeEnabled => !string.IsNullOrEmpty(BuildDeployPreferences.BuildDirectory);
+
+        private static bool ShouldBuildSLNBeEnabled => !isBuilding &&
+                                                       !UwpAppxBuildTools.IsBuilding &&
+                                                       !BuildPipeline.isBuildingPlayer &&
+                                                       !string.IsNullOrEmpty(BuildDeployPreferences.BuildDirectory);
+
+        private static bool ShouldBuildAppxBeEnabled => ShouldBuildSLNBeEnabled && !string.IsNullOrEmpty(BuildDeployPreferences.BuildDirectory);
+
+        private static bool DevicePortalConnectionEnabled => (portalConnections.Connections.Count > 1 || IsHoloLensConnectedUsb) &&
+                                                             !string.IsNullOrEmpty(BuildDeployPreferences.BuildDirectory);
+
+        private static bool CanInstall
+        {
+            get
+            {
+                bool canInstall = true;
+                if (EditorUserBuildSettings.wsaSubtarget == WSASubtarget.HoloLens)
+                {
+                    canInstall = DevicePortalConnectionEnabled && IsHoloLensConnectedUsb;
```

Comment:
I don't think `&& IsHoloLensConnectedUsb` is needed here, since we should support deploying over WiFi as well as over USB.

Reply:
hmm yea but it's actually worse because DevicePortalConnectionEnabled contains IsHoloLensConnectedUsb... I'll look into it